### PR TITLE
Add max and min utils

### PR DESCRIPTION
**Pull request by ZIMkaRU**
### Description:

This PR adds `max` and `min` utils

### New features:
- [x] adds `max` util
- [x] adds `min` util

### PR status:
- [x] Version bumped
- [x] Change-log updated
- [x] Tests added or updated
- [x] PR keeps 100% test coverage
- [x] Type definition updated
- [x] Readme updated
- [x] Linter is applied
- [x] Const arrow functions are used instead of pure function definitions where applicable
- [x] Functions are listed in alphabetic order in index.js, index.d.ts and readme

### CODE REVIEW

File: src/max.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+'use strict'
+
+const determineExtremumValue = require('./util/determineExtremumValue')
+
+const min = (array) => {
```
Comment (dictor93) [inline src/max.js]:
```suggestion
const max = (array) => {
```

File: src/max.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  return determineExtremumValue(
+    array,
+    (value, other) => value > other
+  )
+}
+
+module.exports = min
```
Comment (dictor93) [inline src/max.js]:
```suggestion
module.exports = max
```

File: src/util/determineExtremumValue.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  if (
+    !Array.isArray(array) ||
+    array.length === 0
+  ) {
+    return
+  }
+
+  return array.reduce((prev, curr) => {
+    if (isNil(curr)) {
+      return prev
```
Comment (dictor93) [inline src/util/determineExtremumValue.js:15]: Seems, this condition is not covered by tests

**Review by dictor93 — CHANGES_REQUESTED** [review]
Small comments. The rest LGTM!

File: src/max.js (hunk 1 above)
Comment (ZIMkaRU) [reply src/max.js]: indeed, thanks for typo detection

**Comment by dictor93** [comment]
lgtm

**Review by dictor93 — APPROVED** [review]
LGTM!

**Review by ezewer — APPROVED** [review]
Code looks good to me, please add a test for letters as to prove this also works for text

**Pull request merged**